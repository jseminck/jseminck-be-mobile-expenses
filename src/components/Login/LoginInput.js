import React from 'react';
import { StyleSheet, View, TextInput } from 'react-native';
import Dimensions from 'Dimensions';
import Icon from 'react-native-vector-icons/FontAwesome';

export default class LoginInput extends React.Component {
    static propTypes = {
        icon: React.PropTypes.string.isRequired,
        placeholder: React.PropTypes.string.isRequired,
        password: React.PropTypes.bool,

        // event handlers
        onChangeText: React.PropTypes.func.isRequired
    };

    getInputImageStyles() {
        return [
            styles.inputImage
        ];
    }

    render() {
        const {icon, placeholder, password} = this.props;

        return (
            <View style={styles.input}>
                <Icon
                    name={icon}
                    style={this.getInputImageStyles()}
                />
                <TextInput
                    style={styles.inputField}
                    autoCapitalize='none'
                    placeholder={placeholder}
                    password={password}
                    placeholderTextColor='black'
                    onChangeText={this.props.onChangeText}
                />
            </View>
        );
    }
}

const styles = StyleSheet.create({
    input: {
        flexDirection: 'row',
        marginTop: 30,
        borderWidth: 1,
        borderBottomColor: '#CCC',
        borderColor: 'transparent',
        width: Dimensions.get('window').width
    },
    inputImage: {
        flex: 0.15,
        marginLeft: 50,
        fontSize: 30,
        color: 'black',
        opacity: 0.8
    },
    inputField: {
        flex: 0.85,
        paddingTop: 3,
        height: 30,
        color: 'black'
    }
});
