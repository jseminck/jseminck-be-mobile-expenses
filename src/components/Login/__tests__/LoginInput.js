import 'react-native';
import React from 'react';
import LoginInput from './../LoginInput';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly without password', () => {
    const tree = renderer.create(
        <LoginInput
            icon="glass"
            placeholder="myPlaceholder"
            onChangeText={() => {}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

it('renders correctly with password', () => {
    const tree = renderer.create(
        <LoginInput
            icon="myIcon"
            placeholder="myPlaceholder"
            password={true}
            onChangeText={() => {}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});