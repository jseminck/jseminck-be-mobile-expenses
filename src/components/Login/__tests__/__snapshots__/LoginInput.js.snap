exports[`test renders correctly with password 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#CCC",
      "borderColor": "transparent",
      "borderWidth": 1,
      "flexDirection": "row",
      "marginTop": 30,
      "width": 375
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontFamily": "FontAwesome",
          "fontSize": 12,
          "fontStyle": "normal",
          "fontWeight": "normal"
        },
        Array [
          Object {
            "color": "black",
            "flex": 0.15,
            "fontSize": 30,
            "marginLeft": 50,
            "opacity": 0.8
          }
        ]
      ]
    }>
    ?
  </Text>
  <TextInput
    autoCapitalize="none"
    onChangeText={[Function onChangeText]}
    password={true}
    placeholder="myPlaceholder"
    placeholderTextColor="black"
    style={
      Object {
        "color": "black",
        "flex": 0.85,
        "height": 30,
        "paddingTop": 3
      }
    } />
</View>
`;

exports[`test renders correctly without password 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#CCC",
      "borderColor": "transparent",
      "borderWidth": 1,
      "flexDirection": "row",
      "marginTop": 30,
      "width": 375
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontFamily": "FontAwesome",
          "fontSize": 12,
          "fontStyle": "normal",
          "fontWeight": "normal"
        },
        Array [
          Object {
            "color": "black",
            "flex": 0.15,
            "fontSize": 30,
            "marginLeft": 50,
            "opacity": 0.8
          }
        ]
      ]
    }>
    ï€€
  </Text>
  <TextInput
    autoCapitalize="none"
    onChangeText={[Function onChangeText]}
    password={undefined}
    placeholder="myPlaceholder"
    placeholderTextColor="black"
    style={
      Object {
        "color": "black",
        "flex": 0.85,
        "height": 30,
        "paddingTop": 3
      }
    } />
</View>
`;
